harmony_browse_image_generator/browse.py

```diff
@@ -113,27 +118,39 @@ def create_browse_imagery(
 def convert_mulitband_to_raster(data_array: DataArray) -> ndarray:
     """Convert multiband to a raster image.
 
-    Reads the three/four bands from the file, then normalizes them to the range
+    Reads the three or four bands from the file, then normalizes them to the range
     0 to 255. This assumes the input image is already in RGB or RGBA format and
     just ensures that the output is 8bit.
 
     """
+    if data_array.rio.count not in [3, 4]:
+        raise HyBIGError(
+            f'Cannot create image from {data_array.rio.count} band image. '
+            'Expecting 3 or 4 bands.'
+        )
+
     bands = data_array.to_numpy()
-    norm = Normalize()
 
-    if data_array.rio.count == 3:
-        norm.autoscale(bands)
-        raster = around(norm(bands) * 255.0).astype('uint8')
+    # Create an alpha layer where input NaN values are transparent.
+    nan_mask = np.isnan(bands).any(axis=0)
+    nan_alpha = np.where(nan_mask, TRANSPARENT, OPAQUE)
+
+    # grab any existing alpha layer
+    bands, image_alpha = remove_alpha(bands)
 
-    elif data_array.rio.count == 4:
-        norm.autoscale(bands[0:3, :, :])
-        partial_raster = around(norm(bands[0:3, :, :]) * 255.0).astype('uint8')
-        raster = concatenate([partial_raster.data, bands[3:4, :, :]])
+    norm = Normalize(vmin=np.nanmin(bands), vmax=np.nanmax(bands))
+    raster = np.nan_to_num(np.around(norm(bands) * 255.0), copy=False, nan=0.0).astype(
```

Comment:
nan_to_num: I learned about trying to silence the warning coming from the cast to 'uint8' of NaNs. Rather than swallow the warning decided to be explicit